Cypress targets fail to compile blinky-baremetal in Mbed OS 6

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

Cypress targets fail to compile blinky-baremetal in Mbed OS 6

```
[Fatal Error] cyhal_system.c@30,10: cyabs_rtos.h: No such file or directory
[ERROR] ./mbed-os/targets/TARGET_Cypress/TARGET_PSOC6/psoc6csp/hal/source/cyhal_system.c:30:10: fatal error: cyabs_rtos.h: No such file or directory
   30 | #include "cyabs_rtos.h"

[Fatal Error] scl_buffer_api.h@28,10: 'memp.h' file not found
[ERROR] In file included from ./mbed-os/targets/TARGET_Cypress/TARGET_PSOC6/COMPONENT_SCL/src/scl_buffer_api.c:18:
./mbed-os/targets/TARGET_Cypress/TARGET_PSOC6/COMPONENT_SCL/src/include/scl_buffer_api.h:28:10: fatal error: 'memp.h' file not found
```

#### Target(s) affected by this defect ?

- CY8CKIT_062_BLE
- CY8CPROTO_062_4343W
- CY8CPROTO_062S3_4343W
- CYW9P62S1_43012EVB_01
- CY8CKIT_062_WIFI_BT
- CY8CPROTO_064_SB
- CY8CKIT_064S2_4343W
- CY8CKIT_062S2_43012
- CYW9P62S1_43438EVB_01
- CYSBSYSKIT_01

#### Toolchain(s) (name and version) displaying this defect ?

Arm Compiler 6 and GCC 9

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

mbed-os-6.0.0

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

Mbed CLI 1.10.4

#### How is this defect reproduced ? 

```
mbed import https://github.com/ARMmbed/mbed-os-example-blinky-baremetal
cd mbed-os-example-blinky
mbed compile -t <toolchain> -m <target>
```

@ARMmbed/team-cypress @maclobdell

--- a/rtos/source/TARGET_CORTEX/mbed_lib.json
+++ b/rtos/source/TARGET_CORTEX/mbed_lib.json
@@ -81,6 +81,9 @@
         },
         "NUVOTON": {
             "idle-thread-stack-size-debug-extra": 512
+        },
+        "MCU_PSOC6_M4": {
+            "target.macros_add": ["CY_RTOS_AWARE"]
         }
     }
 }

--- a/targets/targets.json
+++ b/targets/targets.json
@@ -5893,7 +5893,6 @@
         ],
         "macros_add": [
             "MCU_PSOC6_M4",
-            "CY_RTOS_AWARE",
             "CY_USING_HAL",
             "MBED_TICKLESS"
         ],

